### ISSUE TITLE
gc controller run whill be block when the apiservice missing a controller

### ISSUE THREAD

**Issue by fusida** [kubernetes/kubernetes#111354]
### What happened?

if the crd & apiservice not work well, the gc controller will not work.

example:
```
**crd**：challenges.acme.cert-manager.io

**apiservice**：
v1alpha2.acme.cert-manager.io  
v1alpha3.acme.cert-manager.io     
v1beta1.acme.cert-manager.io

**kubectl**:
kubectl get challenges.acme.cert-manager.io  -A
Error from server: conversion webhook for acme.cert-manager.io/v1alpha2, Kind=Challenge failed: Post "[https://cert-manager-webhook.cert-manager.svc:443/convert?timeout=30s](https://cert-manager-webhook.cert-manager.svc/convert?timeout=30s)": service "cert-manager-webhook" not found

**gc controller log**:
garbage controller monitor not yet synced: acme.cert-manager.io/v1beta1, Resource=challenges
```

if gc controller run, will be block to  `garbagecollector.go:147 line`:
```
if !cache.WaitForNamedCacheSync("garbage collector", stopCh, gc.dependencyGraphBuilder.IsSynced) {
    return
}
```
```go
// Run starts garbage collector workers.
func (gc *GarbageCollector) Run(workers int, stopCh <-chan struct{}) {
	defer utilruntime.HandleCrash()
	defer gc.attemptToDelete.ShutDown()
	defer gc.attemptToOrphan.ShutDown()
	defer gc.dependencyGraphBuilder.graphChanges.ShutDown()

	klog.Infof("Starting garbage collector controller")
	defer klog.Infof("Shutting down garbage collector controller")

	go gc.dependencyGraphBuilder.Run(stopCh)

	if !cache.WaitForNamedCacheSync("garbage collector", stopCh, gc.dependencyGraphBuilder.IsSynced) {
		return
	}

	klog.Infof("Garbage collector: all resource monitors have synced. Proceeding to collect garbage")

	// gc workers
	for i := 0; i < workers; i++ {
		go wait.Until(gc.runAttemptToDeleteWorker, 1*time.Second, stopCh)
		go wait.Until(gc.runAttemptToOrphanWorker, 1*time.Second, stopCh)
	}

	<-stopCh
}
```

### What did you expect to happen?

if a crd missing a controller, should not block gc controller run function.

### How can we reproduce it (as minimally and precisely as possible)?

install a crd and apiservice, not install controller operator.

### Kubernetes version

kubernetes 1.20.11

### Cloud provider

<details>
cloud native
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by FillZpp** [comment]
This should be fixed by #110858

**Comment by aojea** [comment]
duplicated of https://github.com/kubernetes/kubernetes/issues/101078, right?
